File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer,Integer> hmap=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            hmap.put(nums[i],hmap.getOrDefault(nums[i],0)+1);
        }
        int num=0;
        for(Map.Entry<Integer,Integer> e:hmap.entrySet()){
            if(e.getValue()>(nums.length/2))
            num=e.getKey();
        }
        return num;
    }
}

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    const hash={};
    let ans=0;
    let maj=0;
    for(let num of nums){
        hash[num]=1+(hash[num]||0);
        if(hash[num]>maj)
        {
            maj=hash[num];
            ans=num;
        }
    }
    return ans;
};